src/promptflow/promptflow/executor/_script_executor.py

```diff
@@ -54,8 +55,12 @@ def exec_line(
         **kwargs,
     ) -> LineResult:
         run_id = run_id or str(uuid.uuid4())
-        with self._update_operation_context(run_id, index):
-            return self._exec_line(inputs, index, run_id, allow_generator_output=allow_generator_output)
+        log_manager = NodeLogManager()
```

Comment:
Add a TODO: we need to extract this to a more general class instead of "NodeLogManager". In this scenario there is no "node" concept.